### PR TITLE
chore: make docker filelock project specific

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
This PR will make the build time docker file lock project-specific instead of system-wide which will allow users to leverage concurrency.

### CODE REVIEW

File: packages/cli/src/commands/build.ts [hunk 1]
```diff
@@ -265,7 +268,7 @@ export default {
             (key.name == "c" && key.ctrl)
           ) {
             await watcher.stop();
-            await dockerLock.release();
+            // await dockerLock.release();
```
Comment (dOrgJelli) [inline packages/cli/src/commands/build.ts]: This is needed. If the process is `kill()`ed, then the file lock will still exist, causing subsequent calls to halt indefinitely.

File: packages/cli/src/lib/project/Web3ApiProject.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export class Web3ApiProject extends Project<Web3ApiManifest> {
     this._buildManifest = undefined;
     this._metaManifest = undefined;
     this._defaultBuildManifestCached = false;
-    this.resetCache();
+    this.removeCacheDir(cacheLayout.buildEnvDir);
```
Comment (dOrgJelli) [inline packages/cli/src/lib/project/Web3ApiProject.ts]: Nice catch :)

File: packages/cli/src/lib/system/file-lock.ts [hunk 3]
```diff
@@ -1,4 +1,17 @@
 import * as fs from "fs";
+import path from "path";
+
+export function createDirs(dirPath: string): void {
```
Comment (dOrgJelli) [inline packages/cli/src/lib/system/file-lock.ts]: This can be done via `fs.mkdirSync(dir, { recursive: true });`

**Review by dOrgJelli — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/build.ts (hunk 1 above)
Comment (Niraj-Kamdar) [reply packages/cli/src/commands/build.ts]: That's just debugging leftover

Comment (Niraj-Kamdar) [reply packages/cli/src/commands/build.ts]: Thanks for catching it

File: packages/cli/src/lib/system/file-lock.ts (hunk 3 above)
Comment (Niraj-Kamdar) [reply packages/cli/src/lib/system/file-lock.ts]: Nice! didn't know about that 😛

**Review by krisbitney — DISMISSED** [review]
LGTM

**Pull request merged**